mafContrib/TDEsPopMetric.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+from lsst.sims.maf.utils import m52snr
+import lsst.sims.maf.metrics as metrics
+import os
+from lsst.sims.utils import uniformSphere
+import lsst.sims.maf.slicers as slicers
+import glob
+from lsst.sims.photUtils import Sed
+
+
+__all__ = ['tde_lc', 'TDEsPopMetric', 'TDEsPopSlicer']
+
+
+class tde_lc(object):
+    """
+    Read in some TDE lightcurves
+
+    Parameters
+    ----------
+    file_list : list of str (None)
+        List of file paths to load. If None, loads up all the files from data/tde/
+    """
+
+    def __init__(self, file_list=None):
+
+        if file_list is None:
+            sims_maf_contrib_dir = os.getenv("SIMS_MAF_CONTRIB_DIR")
+            file_list = glob.glob(os.path.join(sims_maf_contrib_dir, 'data/tde/*.dat'))
+
+        lcs = []
+        for filename in file_list:
+            lcs.append(np.genfromtxt(filename, dtype=[('ph', 'f8'), ('mag', 'f8'), ('filter', 'U1')]))
+
+        # Let's organize the data in to a list of dicts for easy lookup
+        self.data = []
+        filternames = 'ugrizy'
+        for lc in lcs:
+            new_dict = {}
+            for filtername in filternames:
+                infilt = np.where(lc['filter'] == filtername)
+                new_dict[filtername] = lc[infilt]
+            self.data.append(new_dict)
+
+    def interp(self, t, filtername, lc_indx=0):
+        """
+        t : array of floats
+            The times to interpolate the light curve to.
+        filtername : str
+            The filter. one of ugrizy
+        lc_index : int (0)
+            Which file to use.
+        """
+
+        result = np.interp(t, self.data[lc_indx][filtername]['ph'],
+                           self.data[lc_indx][filtername]['mag'],
+                           left=99, right=99)
+        return result
+
+
+class TDEsPopMetric(metrics.BaseMetric):
+    def __init__(self, metricName='TDEsPopMetric', mjdCol='observationStartMJD', m5Col='fiveSigmaDepth',
+                 filterCol='filter', nightCol='night', ptsNeeded=2, file_list=None, mjd0=59853.5,
+                 **kwargs):
+        maps = ['DustMap']
+        self.mjdCol = mjdCol
+        self.m5Col = m5Col
+        self.filterCol = filterCol
+        self.nightCol = nightCol
+        self.ptsNeeded = ptsNeeded
+
+        self.lightcurves = tde_lc(file_list=file_list)
+        self.mjd0 = mjd0
+
+        waveMins = {'u': 330., 'g': 403., 'r': 552., 'i': 691., 'z': 818., 'y': 950.}
+        waveMaxes = {'u': 403., 'g': 552., 'r': 691., 'i': 818., 'z': 922., 'y': 1070.}
```

Comment:
Where did you pull these from, and do you want to provide a utility for updates if/when throughputs changes?